services/userService.js

```diff
@@ -0,0 +1,105 @@
+const bcrypt = require('bcrypt')
+const jwt = require('jsonwebtoken')
+const userDao = require('../models/userDao')
+const checkValidation = require('../utils/validation')
+
+const secretKey = process.env.SECRET_KEY
+
+const createToken = (userInfo) => {
+  const time = new Date().getTime() / 1000 + 60 * 60 * 9
+  const currentTime = Math.floor(time)
+  const expireTime = currentTime + 60 * 60 * 24
+
+  const payload = {
+    userId: userInfo,
+    iss: 'Hyggesil',
+    iat: currentTime,
+    exp: expireTime,
+  }
+  return jwt.sign(payload, secretKey)
+}
+
+const signup = async (name, email, password, phoneNumber) => {
+  const userData = await userDao.checkUserByKakaoIdOrEmail(email)
+  const existsUser = userData[0]
+
+  if (Number(existsUser)) {
+    const err = new Error('ALREADY_EXIST_USER')
+    err.statusCode = 400
+    throw err
+  }
+
+  if (phoneNumber) {
+    const userdata = await userDao.checkUserByphoneNumber(phoneNumber)
+    const existsPhoneNumber = userdata[0]
+    if (Number(existsPhoneNumber)) {
+      const err = new Error('ALREADY_EXIST_PHONE_NUMBER')
+      err.statusCode = 400
+      throw err
+    }
+  }
+
+  await checkValidation.checkValidationEmail(email)
+  await checkValidation.checkValidationPassword(password)
+
+  const hasedPassword = await bcrypt.hash(password, 12)
+  return await userDao.createUser(name, email, hasedPassword, phoneNumber)
+}
+
+const login = async (email, password) => {
+  const userInfo = await userDao.getPasswordByEmail(email)
+
+  if (!userInfo) {
+    const err = new Error('WRONG_EMAIL')
+    err.statusCode = 400
+    throw err
+  }
+
+  const hasedPassword = userInfo.password
+  const checkHash = await bcrypt.compare(password, hasedPassword)
+
+  if (!checkHash) {
+    const err = new Error('WRONG_PASSWORD')
+    err.statusCode = 400
+    throw err
+  }
+
+  const userId = await userDao.getUserId(email)
+  const accesstoken = createToken(userId)
+  return accesstoken
+}
+
+const kakaoLogin = async (kakaoAccessToken) => {
+  const url = 'https://kapi.kakao.com/v2/user/me'
+  const options = {
+    method: 'POST',
+    headers: {
+      Authorization: `Bearer ${kakaoAccessToken}`,
+      'Content-type': 'application/x-www-form-urlencoded;charset=utf-8',
+    },
+  }
+
+  const data = await fetch(url, options)
```

Comment:
fetch는 주로 브라우저에서 request 요청할 때 사용합니다.
node에서는 axios를 더 많이 사용하니 axios로 바꿔주세요.

```suggestion
  const data = await axios(url, options)
```